subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -496,12 +496,8 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
 			}
 			break;
 		case 'p':
-			if (secure_connection) {
```

Comment:
Is this change related to option ordering? This should be reflected in the commit message, or even better put in a separate commit with the fix description in the message.

Reply:
There is no longer a requirement for ordered options with this commit. The previous example would fail to set the PSK but allow the ap enable to proceed on mis ordered options. It now only warns if the PSK is set without the proper security. getopts allows us to not require an order to the options.

Reply:
Should a configuration that has a PSK but does not specify the type of security succeed or fail?

Reply:
FYI, we already do support setting either password (0-63) or SAE passphrase (no length limits). But this isn't exposed to user, it would still use the same `-p` arg, just the validation varies as per `-k`, so, can probably fail here.